Wikipedia/UI-V5/WMFArticleHeaderImageGalleryViewController.m

```diff
@@ -95,45 +98,141 @@ - (UICollectionViewCell*)collectionView:(UICollectionView*)collectionView
         cell.imageView.contentMode = UIViewContentModeScaleAspectFill;
         return cell;
     }
+
     MWKImage* imageMetadata = self.images[indexPath.item];
+
+    // try to set cell's image from memory cache, otherwise fetch it and do face detection if needed
+    if (![self setCachedImageForCell:cell atIndexPath:indexPath metadata:imageMetadata]) {
+        [self fetchImage:imageMetadata forCellAtIndexPath:indexPath];
```

Comment:
But `[self setCachedImageForCell:cell atIndexPath:indexPath metadata:imageMetadata]` could be returning NO because the Face detection didn't run - so in that case, you should run face detection, not fetch the image right?